### PR TITLE
Include AccessDenied key for telemetry event

### PR DESCRIPTION
**Pull request by jennyf19**
Include more info in exception result in xamarin test app

### CODE REVIEW

File: msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs [hunk 1]
```diff
@@ -114,6 +114,7 @@ protected override string GetUIBehaviorPromptValue()
                         AuthenticationRequestParameters.RequestContext)
                         .ConfigureAwait(false);
                 uiEvent.UserCancelled = _authorizationResult.Status == AuthorizationStatus.UserCancel;
+                uiEvent.AccessDenied = _authorizationResult.Status == AuthorizationStatus.ProtocolError;
```
Comment (jennyf19) [inline msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs:117]: when I was testing embedded webview, w/corp account and getting the protocolError, it was getting registered as a UserCancelled, even though it was AccessDenied in the telemetry. I added in this line to register the protocolError.

File: msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs [hunk 2]
```diff
@@ -211,6 +211,8 @@ public void NoCacheLookup()
                 anEvent[EventBase.EventNameKey].EndsWith("ui_event") && anEvent[UiEvent.UserCancelledKey] == "false"));
             Assert.IsNotNull(_myReceiver.EventsReceived.Find(anEvent =>  // Expect finding such an event
                 anEvent[EventBase.EventNameKey].EndsWith("api_event") && anEvent[ApiEvent.UiBehaviorKey] == "select_account"));
+            Assert.IsNotNull(_myReceiver.EventsReceived.Find(anEvent =>  // Expect finding such an event
+                anEvent[EventBase.EventNameKey].EndsWith("ui_event") && anEvent[UiEvent.AccessDeniedKey] == "false"));
```
Comment (jennyf19) [inline msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs:215]: Added `UiEvent.AccessDeniedKey` to preexisting unit tests

File: msal/tests/dev apps/XForms/XForms/AcquirePage.xaml.cs [hunk 3]
```diff
@@ -160,7 +160,7 @@ private string GetSelectedUserId()
             }
             catch (MsalException exception)
             {
-                acquireResponseLabel.Text = "MsalException - " + exception.Message;
+                acquireResponseLabel.Text = String.Format("MsalException -\nError Code: {0}\nMessage: {1}", exception.ErrorCode, exception.Message);
```
Comment (jennyf19) [inline msal/tests/dev apps/XForms/XForms/AcquirePage.xaml.cs:163]: The Xamarin app was just showing `MsalServiceException`, so now it returns a little more info. not all the exceptions have a message.

File: msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs (hunk 1 above)
Comment (henrik-me) [reply msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs:117]: Why call it access denied if it's a protocol error?

---
In reply to: [200799703](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1114#discussion_r200799703) [](ancestors = 200799703)

File: msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs (hunk 2 above)
Comment (henrik-me) [reply msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs:215]: are there any real unit tests for this class? It should be relatively simple to validate.
also seems like this integration test only validated that it's false (not set). Ideally the integration testing would also show an error situation.

---
In reply to: [200799705](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1114#discussion_r200799705) [](ancestors = 200799705)

File: msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs (hunk 1 above)
Comment (jennyf19) [reply msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs:117]: The error that is returned is `access_denied`, and we equate it with an `AuthorizationStatus.ProtocolError` [here](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/core/src/UI/AuthorizationResult.cs#L114), but seems like `access_denied` is more specific, which is why I included that for the telemetry, but can change it to protocol error.

File: msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs (hunk 2 above)
Comment (jennyf19) [reply msal/tests/Test.MSAL.NET.Unit/RequestsTests/InteractiveRequestTests.cs:215]: I added in more test coverage for the uiEvent I added. There are Telemetry tests in core.

**Review by henrik-me — APPROVED** [review]
:shipit:

**Pull request merged**